uint32 frame_count
uint32 diode_count